http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/AwaitCloseChannelPoolMap.java

```diff
@@ -162,42 +166,43 @@ private Bootstrap createBootstrap(URI poolKey) {
     }
 
 
-    private boolean useProxyForHost(URI remoteAddr) {
+    private boolean shouldUseProxyForHost(URI remoteAddr) {
         if (proxyConfiguration == null) {
             return false;
         }
 
-        String remoteHost = StringUtils.lowerCase(remoteAddr.getHost());
-        for (String hostPattern : proxyConfiguration.nonProxyHosts()) {
-            if (remoteHost.matches(hostPattern)) {
-                return false;
-            }
-        }
 
-        return true;
+        return shouldProxyForHostCache.computeIfAbsent(remoteAddr, (uri) ->
```

Comment:
If this needs to be threadsafe you should use a ConcurrentHashMap instead of a HashMap. It will be difficult to write a unit test that asserts brokenness though.

Reply:
Woops good point!